registry/endpoint.go

```diff
@@ -265,18 +262,6 @@ HeaderLoop:
 }
 
 func (e *Endpoint) HTTPClient() *http.Client {
-	tlsConfig := tls.Config{
-		MinVersion: tls.VersionTLS10,
-	}
-	if !e.IsSecure {
-		tlsConfig.InsecureSkipVerify = true
-	}
-	return &http.Client{
-		Transport: &http.Transport{
-			DisableKeepAlives: true,
-			Proxy:             http.ProxyFromEnvironment,
-			TLSClientConfig:   &tlsConfig,
-		},
-		CheckRedirect: AddRequiredHeadersToRedirectedRequests,
-	}
+	// TODO: shouldn't this be ConnectTimeout for safety?
+	return HTTPClient(DefaultTransport(NoTimeout, e.IsSecure))
```

Comment:
Oh well, I don't see why this would be an issue and I do prefer it. Hopefully that resolves https://github.com/docker/docker/pull/13265#discussion_r30450466